tests/test_fits.py

```diff
@@ -84,182 +51,93 @@ def test_setting_arrays_on_fits():
-        dm.dq = np.empty((10, 50, 50), dtype=np.uint32)
+        dm.dq = np.empty((10,), dtype=np.uint32)
 
 
-def delete_array():
-    with pytest.raises(AttributeError):
-        from astropy.io import fits
-        hdulist = fits.HDUList()
-        data = np.empty((50, 50))
-        science = fits.ImageHDU(data=data, name='SCI')
-        hdulist.append(science)
-        hdulist.append(science)
-        with open(hdulist) as dm:
-            del dm.data
-            assert len(hdulist) == 1
-
-
-@pytest.mark.skip("requires RampModel")
-def test_from_fits():
-    with RampModel(FITS_FILE) as dm:
-        assert dm.meta.instrument.name == 'MIRI'
-        assert dm.shape == (5, 35, 40, 32)
+def test_from_scratch(tmp_path):
+    file_path = tmp_path/"test.fits"
 
-
-@pytest.mark.skip("requires ImageModel")
-def test_from_scratch():
-    with ImageModel((50, 50)) as dm:
+    with FitsModel((50, 50)) as dm:
         data = np.asarray(np.random.rand(50, 50), np.float32)
         dm.data[...] = data
 
-        dm.meta.instrument.name = 'NIRCAM'
+        dm.meta.telescope = "EYEGLASSES"
 
-        dm.to_fits(TMP_FITS, overwrite=True)
+        dm.to_fits(file_path)
 
-        with ImageModel.from_fits(TMP_FITS) as dm2:
+        with FitsModel.from_fits(file_path) as dm2:
             assert dm2.shape == (50, 50)
-            assert dm2.meta.instrument.name == 'NIRCAM'
+            assert dm2.meta.telescope == "EYEGLASSES"
             assert dm2.dq.dtype.name == 'uint32'
             assert np.all(dm2.data == data)
 
 
-@pytest.mark.skip("base DataModel no longer has an instrument.name field")
-def test_delete():
-    with DataModel(FITS_FILE) as dm:
-        dm.meta.instrument.name = 'NIRCAM'
-        assert dm.meta.instrument.name == 'NIRCAM'
-        del dm.meta.instrument.name
-        assert dm.meta.instrument.name is None
-
-
-# def test_date_obs():
-#     with DataModel(FITS_FILE) as dm:
-#         assert dm.meta.observation.date.microsecond == 314592
-
-
-@pytest.mark.skip("schemas are not available in this package")
-def test_fits_without_sci():
-    from astropy.io import fits
-    schema = {
-        "allOf": [
-            mschema.load_schema(
-                os.path.join(os.path.dirname(__file__),
-                             "../schemas/core.schema.yaml"),
-                resolve_references=True),
-            {
-                "type": "object",
-                "properties": {
-                    "coeffs": {
-                        'max_ndim': 1,
-                        'fits_hdu': 'COEFFS',
-                        'datatype': 'float32'
-                    }
-                }
-            }
-        ]
-    }
-
-    fits = fits.HDUList(
-        [fits.PrimaryHDU(),
-         fits.ImageHDU(name='COEFFS', data=np.array([0.0], np.float32))])
-
-    with DataModel(fits, schema=schema) as dm:
-        assert_array_equal(dm.coeffs, [0.0])
-
-
 def _header_to_dict(x):
     return dict((a, b) for (a, b, c) in x)
```

Comment:
Is this needed anymore now that it has been removed from `test_extra_fits`?

Reply:
No sir!  I shall remove it.